client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -120,19 +122,31 @@ module.exports = React.createClass( {
 
 			if ( cartItems.hasRenewalItem( this.state.previousCart ) ) {
 				renewalItem = cartItems.getRenewalItems( this.state.previousCart )[ 0 ];
+
 				redirectTo = purchasePaths.managePurchase( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId );
 			}
 		}
 
 		page.redirect( redirectTo );
+
 		return true;
 	},
 
 	getCheckoutCompleteRedirectPath: function() {
 		var renewalItem;
+
 		if ( cartItems.hasRenewalItem( this.props.cart ) ) {
 			renewalItem = cartItems.getRenewalItems( this.props.cart )[ 0 ];
+
 			return purchasePaths.managePurchaseDestination( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId, 'thank-you' );
+		} else if ( cartItems.hasFreeTrial( this.props.cart ) ) {
```

Comment:
I agree with @Tug's concerns here, but I think we should deal with them in a different PR. @Tug are you ok to open an issue for this?